genesyscloud/resource_genesyscloud_telephony_providers_edges_trunk.go

```diff
@@ -0,0 +1,202 @@
+		if err != nil {
+			return diag.Errorf("Failed to assign trunk base settings to edge group %s: %s", edgeGroupId, err)
+		}
+	} else {
+		return diag.Errorf("edge_id or edge_group_id were not set. One must be set in order to assign the trunk base settings")
+	}
+
+	trunk, err := getTrunkByTrunkBaseId(trunkBaseId, meta)
+	if err != nil {
+		return diag.Errorf("Failed to get trunk by trunk base id %s: %s", trunkBaseId, err)
+	}
+
+	d.SetId(*trunk.Id)
+
+	log.Printf("Created trunk %s", *trunk.Id)
+
+	return readTrunk(ctx, d, meta)
+}
+
+func getTrunkByTrunkBaseId(trunkBaseId string, meta interface{}) (*platformclientv2.Trunk, error) {
+	sdkConfig := meta.(*providerMeta).ClientConfig
+	edgesAPI := platformclientv2.NewTelephonyProvidersEdgeApiWithConfig(sdkConfig)
+
+	// It should return the trunk as the first object. Paginating to be safe
+	for pageNum := 1; ; pageNum++ {
+		trunks, _, getErr := edgesAPI.GetTelephonyProvidersEdgesTrunks(pageNum, 100, "", "", "", trunkBaseId, "")
+		if getErr != nil {
+			return nil, fmt.Errorf("Failed to get page of edge groups: %v", getErr)
+		}
+
+		if trunks.Entities == nil || len(*trunks.Entities) == 0 {
+			break
+		}
+
+		for _, trunk := range *trunks.Entities {
+			if *trunk.Id != trunkBaseId {
+				return &trunk, nil
+			}
+		}
+	}
+
+	return nil, fmt.Errorf("Could not find trunk for trunk base setting id: %v", trunkBaseId)
+}
+
+func updateTrunk(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	return createTrunk(ctx, d, meta)
+}
+
+func readTrunk(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*providerMeta).ClientConfig
+	edgesAPI := platformclientv2.NewTelephonyProvidersEdgeApiWithConfig(sdkConfig)
+
+	log.Printf("Reading trunk %s", d.Id())
+	trunk, resp, getErr := edgesAPI.GetTelephonyProvidersEdgesTrunk(d.Id())
+	if getErr != nil {
+		if resp != nil && resp.StatusCode == 404 {
+			d.SetId("")
+			return nil
+		}
+		return diag.Errorf("Failed to read trunk %s: %s", d.Id(), getErr)
+	}
+
+	log.Printf("Read trunk %s %s", d.Id(), *trunk.Name)
+
+	return nil
+}
+
+func deleteTrunk(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	// Does not delete the trunk. This resource will just no longer manage trunks.
+	return nil
+}
+
+func trunkExporter() *ResourceExporter {
+	return &ResourceExporter{
+		GetResourcesFunc: getAllWithPooledClient(getAllTrunks),
+		RefAttrs: map[string]*RefAttrSettings{
+			"trunk_base_settings_id": {RefType: "genesyscloud_telephony_providers_edges_trunkbasesettings"},
+			"edge_group_id":          {RefType: "genesyscloud_telephony_providers_edges_edge_group"},
+		},
+	}
+}
+
+func getAllTrunks(ctx context.Context, sdkConfig *platformclientv2.Configuration) (ResourceIDMetaMap, diag.Diagnostics) {
+	resources := make(ResourceIDMetaMap)
+
+	edgesAPI := platformclientv2.NewTelephonyProvidersEdgeApiWithConfig(sdkConfig)
+
+	for pageNum := 1; ; pageNum++ {
```

Comment:
Why not use a range instead of this.  ^^^^

Reply:
Yeah that could be done using a do while. For consistency it should be done to all resources if we are to go that route but there's not much need since the loop breaks when there are no more entities